### PR TITLE
[Update] 3.4.4

### LINKED ISSUE #317: [Command Bug] `[p]backpack sell` doesn't always release a user lock
## Complete these first
- [x] I confirm that I am the owner of the bot running Adventure.
- [x] I confirm that I have not modified the source code of Adventure.
- [ ] I'm an Adventure user but not the bot owner.
- [x] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Command name

`[p]backpack sell`

## Expected Behavior
I expect users to always be able to use adventure commands after using `[p]backpack sell`.

## Current Behavior
Currently there's an issue with the command not always releasing the lock.
Adventure version: 3.4.3

## Possible Solution
I suggest rewriting how you handle the locks in that particular command and wrap all these lines in `async with lock:`:
https://github.com/aikaterna/gobcog/blob/master/adventure/adventure.py#L1461-L1504
And of course remove any other lock releasing there as well (including the one in `_backpack_sell_button_action`) since it would then be handled by context manager. And while alternatively you could do a simpler fix which is just adding lock releasing [here (L1479)](https://github.com/aikaterna/gobcog/blob/master/adventure/adventure.py#L1479), I think it's a worse and more error-prone solution.

## Steps to Reproduce
I'm sorry, I haven't used Adventure personally in quite a while, so I am not entirely sure how can the code I linked be reached, but I think that it's pretty clear what's wrong from my description above.
If you don't agree though, let me know and I'll try to add reproduction steps.

## Context (Environment)
This issue rendered one of my users unable to use any adventure commands after using `[p]backpack sell`

### LINKED ISSUE #321: [Bug] Running does not actually decreases money
- [X] I confirm that I am the owner of the bot running Adventure.
- [X] I confirm that I have not modified the source code of Adventure.
- [X] I'm an Adventure user but not the bot owner. (State, in which bug occurred)
- [X] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Expected Behavior
Running withdraws money

## Current Behavior
Bot says that money is used, but it is still in bank

## Steps to Reproduce
1. Check your balance
2. Wait until someone else start an adventure
3. Run from it
4. Wait for adventure to end
5. Check your balance again

img
img
img

### LINKED ISSUE #329: [Bug] Can't Equip Item Bought from Cart
## Complete these first
- [x] I confirm that I am the owner of the bot running Adventure.
- [x] I confirm that I have not modified the source code of Adventure.
- [ ] I'm an Adventure user but not the bot owner.
- [x] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Expected Behavior
Item to be equipped.

## Current Behavior
It doesn't find the item.

## Steps to Reproduce
No idea.

## Context (Environment)
Equip my item I just bought from the cart.

Other two items are recognized.
image

### LINKED ISSUE #330: [Command Bug] backpack trade accepts negative `asking` value
## Complete these first
- [X] I confirm that I am the owner of the bot running Adventure.
- [X] I confirm that I have not modified the source code of Adventure.
- [ ] I'm an Adventure user but not the bot owner.
- [X] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Command name

`backpack trade`

## Expected Behavior
Command send some sort of error message before even try to trade

## Current Behavior
Command proceeds to trade, and throws traceback on confirmation.

## Possible Solution
Check for negative values of `asking` value

## Steps to Reproduce
1. Use `[p]backpack trade @DiscordUser#0000 -1 name_of_item_from_backpack`
2. Wait until `@DiscordUser#0000` confirms trade
3. Check your logs

### Traceback
```
[2020-09-25 11:57:02] [ERROR] red: ValueError
Traceback (most recent call last):
  File "/home/fixator/Red-V3/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/home/fixator/.local/share/Red-DiscordBot/cogs/CogManager/cogs/adventure/adventure.py", line 1726, in backpack_trade
    await bank.transfer_credits(buyer, ctx.author, asking)
  File "/home/fixator/.local/share/Red-DiscordBot/cogs/CogManager/cogs/adventure/bank.py", line 325, in transfer_credits
    return await bank.transfer_credits(from_=from_, to=to, amount=amount)
  File "/home/fixator/Red-V3/lib/python3.8/site-packages/redbot/core/bank.py", line 428, in transfer_credits
    raise ValueError(
ValueError: Invalid transfer amount -5 <= 0
```

## Possible Implementation
```py
if asking < 0:
    return await ctx.send("You can't *sell* for less than 0...")
```

### LINKED ISSUE #331: [Bug] Error in formatting the rebirth box
## Complete these first
- [ ] I confirm that I am the owner of the bot running Adventure.
- [x] I confirm that I have not modified the source code of Adventure.
- [x] I'm an Adventure user but not the bot owner.
- [x] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Steps to Reproduce

1.
2.
3.
4.
image

### LINKED ISSUE #332: [Bug] Cart should not appear while there is an active adventure
## Complete these first
- [x] I confirm that I am the owner of the bot running Adventure.
- [x] I confirm that I have not modified the source code of Adventure.
- [ ] I'm an Adventure user but not the bot owner.
- [x] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Expected Behavior

I should be busy with an adventure and not be distracted by a cart appearing.
## Current Behavior

version 3.4.3
I started an adventure and part way through the adventure a cart appeared distracting everyone from the ongoing adventure. People couldn't buy anything from the cart until the adventure is complete either.

## Possible Solution

Prevent cart from appearing at all if there's a current adventure in the guild.
## Steps to Reproduce

1. Start adventure
2. Somtimes cart appears
3. ???
4. Profit (but actually no profit cause can't use cart)

## Context (Environment)

Was trying to enjoy an adventure and the cart kept getting in the way and people couldn't interact with it anyways since they already subscribed to the adventure.

Start an adventure, cart appears no one can participate in cart while adventure is happening.
## Detailed Description

How many times do I need to restate that a cart can appear while an adventure is happening?
## Possible Implementation

Stop carts from appearing while an adventure is currently active in the guild.

### LINKED ISSUE #337: [Bug] Spending Skill Points Response
## Complete these first
- [x] I confirm that I am the owner of the bot running Adventure.
- [x] I confirm that I have not modified the source code of Adventure.
- [ ] I'm an Adventure user but not the bot owner.
- [ ] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

User wanted to spend their 2 skill points but typed 3 by mistake. Maybe a better response when trying to skill up with more points than you own.
image

### LINKED ISSUE #339: [Command Bug] cbackpack does not show Tinkerers' forged items
## Complete these first
- [ ] I confirm that I am the owner of the bot running Adventure.
- [ ] I confirm that I have not modified the source code of Adventure.
- [x] I'm an Adventure user but not the bot owner.
- [x] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Command name

`cbackpack show` - this is related to the command group

## Expected Behavior
Using `cbackpack show` with any flag (for example just `--diff`) should show Tinkerer's forged item, like every other item.

## Current Behavior
The command instead acts as if the forge item was not in backpack at all - doesn't show up in `--match` searches, and the `forged` rarity is not listed as an option when viewing the help message for the `--rarity` flag.

## Possible Solution
The `forged` rarity may be ignored, or not taken into account by the `cbackpack` commands.

## Steps to Reproduce
1. Get a Forged item as Tinkerer class.
2. Run `cbackpack --diff` or any other filter that matches the forged item.

## Context (Environment)
I was trying to specifically look at my forged item's stats, while it was unequipped. This happened on Cat, in Red's discord.

## Detailed Description
-

## Possible Implementation
-

### LINKED ISSUE #343: Reaction Lag? [adv]
Sometimes adventure doesn’t recognize reactions even though the timer is anywhere from 30 second to 3 minutes remaining.

### LINKED ISSUE #344: [Command Bug] Editing/add reacts to missing messages
## Complete these first
- [x] I confirm that I am the owner of the bot running Adventure.
- [x] I confirm that I have not modified the source code of Adventure.
- [ ] I'm an Adventure user but not the bot owner.
- [x] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Command name
negaverse and/or any other commands that deal with edits, and commands or actions that add reacts to messages

## Expected Behavior
return/pass on deleted messages or attempts to add reacts

## Current Behavior
adventure version 3.4.3.1
a couple different examples
```
[2021-01-08 22:28:21] [ERROR] red: Exception in command 'negaverse'
Traceback (most recent call last):
  File "/.../.local/share/cat38/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/.../.local/share/Red-DiscordBot/cogs/CogManager/cogs/adventure/adventure.py", line 4040, in _negaverse
    await nv_msg.edit(content=entry_msg)
  File "/.../.local/share/cat38/lib/python3.8/site-packages/discord/message.py", line 884, in edit
    data = await self._state.http.edit_message(self.channel.id, self.id, **fields)
  File "/.../.local/share/cat38/lib/python3.8/site-packages/discord/http.py", line 243, in request
    raise NotFound(r, data)
discord.errors.NotFound: 404 Not Found (error code: 10008): Unknown Message

[2021-01-08 22:31:53] [CRITICAL] red.main: Caught unhandled exception in task:
Traceback (most recent call last):
  File "/.../.local/share/Red-DiscordBot/cogs/CogManager/cogs/adventure/menus.py", line 491, in add_reactions_task
    await msg.add_reaction(emoji)
  File "/.../.local/share/cat38/lib/python3.8/site-packages/discord/message.py", line 994, in add_reaction
    await self._state.http.add_reaction(self.channel.id, self.id, emoji)
  File "/.../.local/share/cat38/lib/python3.8/site-packages/discord/http.py", line 243, in request
    raise NotFound(r, data)
discord.errors.NotFound: 404 Not Found (error code: 10008): Unknown Message
```

## Possible Solution
I realize all catching all edits/add reacts may be a pain in the ass so maybe this is a non-issue

## Steps to Reproduce
1. Use negaverse
2. Have some method of deleting the message like another bot cleaning the channel
3. Bot tries to edit message that doesn't exist
4. :pandaexplode:

## Context (Environment)
This can interrupt actions and ideally some sort of friendly user facing message talking about the command being interrupted because of not being able to finish the process would be nice in case of a command that can't be completed, and/or making sure that these occasions don't cause cooldown locks, removal of user currency without reward, etc

## Detailed Description
If it's too much of a pain to catch everything, at least audit the path the precursors to these errors take to reduce causes of user frustration/actions consuming cooldowns with no rewards

## Possible Implementation
No suggestions without getting into the code.

### LINKED ISSUE #349: [Feature Request] rebirth level changing
# Feature request

## Complete these first
- [x] I confirm that I have checked the [issue tracker](../) and a this feature request doesn't already exist.

## Select the type of feature you are requesting:

- [x] Functionality
- [ ] Command

## Describe your requested feature

currently if im not mistaken, after rebirth 10, the max level will be at 75 and with addition of 10
always end on a 5

i am proposing changing it to 0, by either making rebirth 10 also add 5, or add 15

i just think numbers ending with 0 looks better

### LINKED ISSUE #352: [Command Bug] backpack eset doesn't work as intended
## Complete these first
- [x] I confirm that I am the owner of the bot running Adventure.
- [x] I confirm that I have not modified the source code of Adventure.
- [ ] I'm an Adventure user but not the bot owner.
- [x] I confirm that I have checked the [issue tracker](../) and a this bug hasn't been reported already.

## Command name

[p]backpack eset

## Expected Behavior
eset should equip your equip-able set items, using the name of set

## Current Behavior
Adventure version: 3.4.3.2

"You have no pieces of {set_name} that you can equip"

## Possible Solution
from Draper [Discord message link](https://discord.com/channels/240154543684321280/735324427302076456/807194319269789696)
...there is a bug in https://github.com/aikaterna/gobcog/blob/master/adventure/adventure.py#L1213, it shouldnt call .title()

## Steps to Reproduce
https://cdn.discordapp.com/attachments/735324427302076456/807187481643581470/Screen_Shot_2021-02-05_at_5.54.46_PM.png
1. have set
2. do [p]backpack eset {set_name}
3. doesn't let you q.q

### PR DESCRIPTION
**Pull request by Drapersniper**
- allow using Adventure with Red 3.3.12 or higher.
- fixes #317
- fixes #349
- fixes #352
- fixes #344
- fixes #321
- fixes #339
- fixes #337
- fixes #329
- fixes #332
- fixes #330
- fixes #331
- fixes #343

### CODE REVIEW

File: adventure/adventure.py [hunk 1]
```diff
@@ -7601,15 +7652,15 @@ async def _reward(self, ctx: commands.Context, userlist, amount, modif, special)
...
+            userxp = int(userxp * (c.gear_set_bonus.get("xpmult", 1) + daymult) * session_bonus)
             usercp = int(usercp * (c.gear_set_bonus.get("cpmult", 1) + daymult))
             newxp += userxp
             newcp += usercp
             roll = random.randint(1, 5)
             if c.heroclass.get("pet", {}).get("bonuses", {}).get("always", False):
                 roll = 5
             if roll == 5 and c.heroclass["name"] == "Ranger" and c.heroclass["pet"]:
-                petxp = int(userxp * c.heroclass["pet"]["bonus"])
+                petxp = int(userxp * c.heroclass["pet"]["bonus"] * session_bonus)
```
Comment (Drapersniper) [inline adventure/adventure.py:7663]:
```suggestion
                petxp = int(userxp * c.heroclass["pet"]["bonus"])
```
Yeeee no ...

File: adventure/adventure.py [hunk 2]
```diff
@@ -7601,7 +7652,7 @@ async def _reward(self, ctx: commands.Context, userlist, amount, modif, special)
                 continue
             userxp = int(xp + (xp * 0.5 * c.rebirths) + max((xp * 0.1 * min(250, c._int / 10)), 0))
             usercp = int(cp + max((cp * 0.1 * min(1000, (c._luck + c._att) / 10)), 0))
-            userxp = int(userxp * (c.gear_set_bonus.get("xpmult", 1) + daymult))
+            userxp = int(userxp * (c.gear_set_bonus.get("xpmult", 1) + daymult) * session_bonus)
```
Comment (Drapersniper) [inline adventure/adventure.py:7655]:
```suggestion
            userxp = int(userxp * (c.gear_set_bonus.get("xpmult", 1) + daymult + session_bonus))
```

File: adventure/adventure.py [hunk 3]
```diff
@@ -7592,6 +7638,11 @@ async def _reward(self, ctx: commands.Context, userlist, amount, modif, special)
         reward_message = ""
         currency_name = await bank.get_currency_name(ctx.guild,)
         can_embed = not ctx.guild or (await _config.guild(ctx.guild).embed() and await ctx.embed_requested())
+        session = self._sessions.get(ctx.guild.id)
+        if session:
+            session_bonus = 1 if session.easy_mode else 2
+        else:
+            session_bonus = 1
```
Comment (Drapersniper) [inline adventure/adventure.py:7645]:
```suggestion
            session_bonus = 0 if session.easy_mode else 1
        else:
            session_bonus = 0
```

**Comment by aikaterna** [comment]
Adventure end message that includes gold and xp summary does not have a closing phrase.

Example:

aikaterna used 334 gold coins,
flare used 4 gold coins,
Twentysix - Soultrader Edition used 5 gold coins, and
MAX used 16,096 gold coins

Should be:

aikaterna used 334 gold coins,
flare used 4 gold coins,
Twentysix - Soultrader Edition used 5 gold coins, and
MAX used 16,096 gold coins to repair their gear.

**Comment by aikaterna** [comment]
A message gets a close emoji that I wouldn't have expected.

[p]backpack sell tiny

Tiny matches multiple items

Multiple items share that name, which one would you like?
0. {Legendary:'Tiny Cobalt Vest Of Fools'} (owned 1)
1. {Legendary:'Tiny Vibranium Band'} (owned 1)
2. {Ascended:''Tiny Platinum Chestplate Of The Mind''} (owned 1)
3. {Ascended:''Tiny Ebony Clogs Of Fury''} (owned 1)
4. {Ascended:''Tiny Iron Gauntlets''} (owned 1)
5. {Ascended:''Tiny Nanite Grips Of The Nimble''} (owned 1)

Select one

On the next dialog box:
"aikaterna, do you want to sell this item for 21,404 each?"
click X 

"Not selling those items" receives an X instead of the item list

Alternatively, maybe just not putting an X on that message at all would be ok if that's a thing

**Comment by aikaterna** [comment]
Heroclass command

Change from tinkerer -> anything

Hit yes for Do you want to continue?
Hit no for Shall I proceed?

No message from the bot saying they won't proceed.

**Comment by Vuks69** [comment]
Having --match case sensitive by default is somewhat counterintuitive imo, as the only items (that I know of) that have their names lowercase only are the tinkerers' forged items. This may be also confusing when a query like `[p]cbackpack show --match cursed` doesn't return anything.

Current behaviour is also inconsistent with `[p]forge` item selection.

**Comment by Drapersniper** [comment]
> Having `cbackpack`'s `--match` arg case sensitive by default is somewhat counterintuitive imo, as the only items (that I know of) that have their names lowercase only are the tinkerers' forged items. This may be also confusing when a query like `[p]cbackpack show --match cursed` doesn't return anything.
> 
> Current behaviour is also inconsistent with `[p]forge` item selection.

Cbackpack is a completely new command, not making it case insensitive by default, if people want it to be case insensitive they can use the flag to do so, just like it's explained in the instructions for the command.

**Comment by Drapersniper** [comment]
https://discord.com/channels/489162733791739950/543214506759815198/751073466760822895

File: adventure/adventure.py [hunk 4]
```diff
@@ -5610,7 +5846,16 @@ async def update_monster_roster(self, user):
     async def _simple(self, ctx: commands.Context, adventure_msg, challenge: str = None, attribute: str = None):
         self.bot.dispatch("adventure", ctx)
         text = ""
+        c = await Character.from_json(self.config, ctx.author, self._daily_bonus)
         easy_mode = await self.config.easy_mode()
+        if not easy_mode:
+            if c.rebirths >= 25:
```
Comment (Drapersniper) [inline adventure/adventure.py:5852]:
```suggestion
            if c.rebirths >= 30:
```

**Comment by Drapersniper** [comment]
fixes #349
fixes #352

**Pull request merged**